### PR TITLE
perf: optimized asset filtering in zrx and cow swappers

### PR DESCRIPTION
**Pull request by woodenfurniture**
## Description

This PR improved the performance of ZRX and Cow swappers to reduce app load time be ~1.24s, and reduces glitching during trade input.

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [x] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

NA

## Risk

- Minor risk of asset filtering broken. This should would at worst display "invalid trade pair" or prevent assets being selected - would not break actual trades

## Testing

- Automated tests updated. 
- Manually tested to check assets are being displayed as expected.

Performance before changes - 1240 milliseconds

Performance after changes - 74 milliseconds

### Operations

- This is user facing
- Test that trade pairs are working as expected against production. Available assets and trade pairs should be identical to production.

### CODE REVIEW

**Comment by woodenfurniture** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4298** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4298" target="_blank">Graphite</a> 
    * **PR #4310** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4310" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4310?utm_source=stack-comment).

File: src/lib/swapper/manager/SwapperManager.test.ts [hunk 1]
```diff
@@ -136,69 +150,53 @@ describe('SwapperManager', () => {
...
     it('should return unique assetIds', () => {
-      const assetIds = [
-        'bip122:000000000019d6689c085ae165831e93/slip44:0',
-        'eip155:1/erc20:0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9', // Aave
-        'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d', // Fox
-        'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d', // Fox (duplicate)
-      ]
+      const assetIds = [BTC.assetId, WETH.assetId, FOX.assetId, FOX.assetId]
 
       const sellAssetId = 'eip155:1/erc20:0xc770eefad204b5180df6a14ee197d99d808ee52d'
```
Comment (gomesalexandre) [inline src/lib/swapper/manager/SwapperManager.test.ts:167]: While at it, can we use BTC or WETH as a duplicate instead and remove FOX.assetId from here? FOX can't be buyAsset for FOX sellAsset
```suggestion
      const sellAssetId = foxAssetId
```

File: src/lib/swapper/swappers/CowSwapper/CowSwapper.ts [hunk 2]
```diff
@@ -90,29 +92,34 @@ export class CowSwapper implements Swapper<KnownChainIds.EthereumMainnet> {
...
   filterBuyAssetsBySellAssetId(args: BuyAssetBySellIdInput): AssetId[] {
     const { assetIds = [], sellAssetId } = args
+    const assets = selectAssets(store.getState())
+    const sellAsset = assets[sellAssetId]
+
     if (
-      fromAssetId(sellAssetId).assetNamespace !== 'erc20' ||
+      sellAsset === undefined ||
+      sellAsset.chainId !== KnownChainIds.EthereumMainnet ||
+      sellAssetId === ethAssetId || // can sell erc20 only
```
Comment (gomesalexandre) [inline src/lib/swapper/swappers/CowSwapper/CowSwapper.ts:101]: Very nice additional safety ser

**Review by gomesalexandre — APPROVED** [review]
Tested locally, noticed no regressions in ZRX and CoW swappers, conceptually looks excellent and improves perceived perf.

Note I didn't actually broadcast Txs, but confirmed with MM the Tx data looked sane.

Added a few comments to be tackled at your convenience.

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**